pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/ZKMetadataUtils.java

```diff
@@ -56,6 +60,32 @@ public static void refreshSegmentZKMetadata(String tableNameWithType, SegmentZKM
         segmentSizeInBytes, false);
   }
 
+  public static void updateSegmentZKMetadataInterval(TableConfig tableConfig, SegmentZKMetadata segmentZKMetadata,
+      Schema oldSchema, Schema newSchema) {
+    String timeColumnName = tableConfig.getValidationConfig().getTimeColumnName();
+    if (StringUtils.isNotEmpty(timeColumnName)) {
+      DateTimeFieldSpec currentDateTimeFieldSpec = oldSchema.getDateTimeSpec(timeColumnName);
+      DateTimeFieldSpec newDateTimeFieldSpec = newSchema.getDateTimeSpec(timeColumnName);
+
+      long startTime = segmentZKMetadata.getStartTimeMs();
```

Comment:
Currently we store the converted start/end time in the ZK metadata (always EPOCH). We should also store the raw min/max time in the ZK metadata, and use them and the schema to re-generate the start/end time.
When the raw min/max time value doesn't comply with the time format, or converted start/end time is not in the right range, we can put `null` as start/end time, and make the caller able to handle it (e.g. skip the segment for time management purpose)